include/swift/Basic/TaskQueue.h

```diff
@@ -39,6 +40,66 @@ enum class TaskFinishedResponse {
   StopExecution,
 };
 
+/// TaskProcessInformation is bound to a task and contains information about the
+/// process that ran this task. This is especially useful to find out which
+/// tasks ran in the same process (in multifile-mode or when WMO is activated
+/// e.g.). If available, it also contains information about the usage of
+/// resources like CPU time or memory the process used in the system. How ever,
+/// this could differ from platform to platform and is therefore optional.
```

Comment:
This is explicitly to not about task resource usage but about the usage of the process that ran this task. But it might be worth mentioning that this is the case. The difference between the two is already becoming very confusing!
A process can run multiple tasks (e.g. a process that compiles 5 swift files) but still all tasks are separately outputted. The usage relates always to the process so if someone would like to sum all usages up, (s)he should unique by `real-pid` first. I'll add this to the comment since it's very essential!
The 'if available' relates to the fact that this is depending on the `rusage` and `wait4` symbols. If they are not available on your system, you also don't get any usage information. You could also run the driver by stating `skip-execution` and you would also get no usage information. I'll also add this to the comment!

Thanks for the talk, really appreciate your feedback! :)

Reply:
Yes, it sure can be confusing.Your comment will be great! My pleasure, I enjoyed it.